pkg/tsdb/azuremonitor/azure-resource-graph-datasource_test.go

```diff
@@ -0,0 +1,77 @@
+package azuremonitor
+
+import (
+	"fmt"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuildingAzureResourceGraphQueries(t *testing.T) {
+	datasource := &AzureResourceGraphDatasource{}
+	fromStart := time.Date(2018, 3, 15, 13, 0, 0, 0, time.UTC).In(time.Local)
+
+	tests := []struct {
+		name                      string
+		queryModel                []plugins.DataSubQuery
+		timeRange                 plugins.DataTimeRange
+		azureResourceGraphQueries []*AzureResourceGraphQuery
+		Err                       require.ErrorAssertionFunc
+	}{
+		{
+			name: "Query with macros should be interpolated",
+			timeRange: plugins.DataTimeRange{
+				From: fmt.Sprintf("%v", fromStart.Unix()*1000),
+				To:   fmt.Sprintf("%v", fromStart.Add(34*time.Minute).Unix()*1000),
+			},
+			queryModel: []plugins.DataSubQuery{
+				{
+					DataSource: &models.DataSource{
+						JsonData: simplejson.NewFromAny(map[string]interface{}{}),
+					},
+					Model: simplejson.NewFromAny(map[string]interface{}{
+						"queryType": "Azure Resource Graph",
+						"azureResourceGraph": map[string]interface{}{
+							"query":        "Resources | project name, properties, '$workspace' | order by name asc",
+							"resultFormat": timeSeries,
+						},
+					}),
+					RefID: "A",
+				},
+			},
+			azureResourceGraphQueries: []*AzureResourceGraphQuery{
+				{
+					RefID:        "A",
+					ResultFormat: "table",
+					URL:          "",
+					Model: simplejson.NewFromAny(map[string]interface{}{
+						"azureResourceGraph": map[string]interface{}{
+							"query":        "Resources | project name, properties, '$workspace' | order by name asc",
+							"resultFormat": timeSeries,
+						},
+					}),
+					Params: url.Values{"query": {""}},
```

Comment:
Isn't the whole idea of this test to verify that macro interpolation works? Now you're not testing that? I'll fix...